modules/openapi-generator/src/main/resources/csharp-netcore/libraries/generichost/ClientUtils.mustache

```diff
@@ -0,0 +1,337 @@
+
+            foreach (var contentType in contentTypes)
+            {
+                if (IsJsonMime(contentType))
+                    return contentType;
+            }
+
+            return contentTypes[0]; // use the first content type specified in 'consumes'
+        }
+
+        /// <summary>
+        /// Select the Accept header's value from the given accepts array:
+        /// if JSON exists in the given array, use it;
+        /// otherwise use all of them (joining into a string)
+        /// </summary>
+        /// <param name="accepts">The accepts array to select from.</param>
+        /// <returns>The Accept header to use.</returns>
+        public static string SelectHeaderAccept(string[] accepts)
+        {
+            if (accepts.Length == 0)
+                return null;
+
+            if (accepts.Contains("application/json", StringComparer.OrdinalIgnoreCase))
+                return "application/json";
+
+            return string.Join(",", accepts);
+        }
+
+        /// <summary>
+        /// Provides a case-insensitive check that a provided content type is a known JSON-like content type.
+        /// </summary>
+        public static readonly Regex JsonRegex = new Regex("(?i)^(application/json|[^;/ \t]+/[^;/ \t]+[+]json)[ \t]*(;.*)?$");
+
+        /// <summary>
+        /// Check if the given MIME is a JSON MIME.
+        /// JSON MIME examples:
+        ///    application/json
+        ///    application/json; charset=UTF8
+        ///    APPLICATION/JSON
+        ///    application/vnd.company+json
+        /// </summary>
+        /// <param name="mime">MIME</param>
+        /// <returns>Returns True if MIME type is json.</returns>
+        public static bool IsJsonMime(string mime)
+        {
+            if (string.IsNullOrWhiteSpace(mime)) return false;
+
+            return JsonRegex.IsMatch(mime) || mime.Equals("application/json-patch+json");
+        }
+
+        /// <summary>
+        /// The base path of the API
+        /// </summary>
+        public const string BASE_ADDRESS = "{{{basePath}}}";
+
+        /// <summary>
+        /// The scheme of the API
+        /// </summary>
+        public const string SCHEME = "{{{scheme}}}";
+
+        /// <summary>
+        /// The context path of the API
+        /// </summary>
+        public const string CONTEXT_PATH = "{{contextPath}}";
+
+        /// <summary>
+        /// The host of the API
+        /// </summary>
+        public const string HOST = "{{{host}}}";
+
+        /// <summary>
+        /// The format to use for DateTime serialization
+        /// </summary>
+        public const string ISO8601_DATETIME_FORMAT = "o";
+
+        /// <summary>
+        /// Add the api to your host builder.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="options"></param>
+        public static IHostBuilder Configure{{packageName}}(this IHostBuilder builder, Action<HostBuilderContext, HostConfiguration> options)
```

Comment:
This doesn't work if your package name is a non trivial namespace, i.e. Company.ApiClients.Twitter or similar.

Reply:
Replaced packageName with projectName and adding regex validation to ensure we can use it for a class name. I saw this projectName variable elsewhere, but didn't see it used in csharp. Let me know if this should change.